src/ZeroLog.Impl.Full/LogManager.AppenderThread.cs

```diff
@@ -85,16 +96,21 @@ private void WriteToAppenders()
                     FlushAppenders();
                     flush = false;
                 }
+                else if (_nextConfig is not null)
```

Comment:
We could maybe try to apply the new configuration every time and if the call to `Interlocked.Exchange` returns `null` fall down to the `else` branch.  I.e. we could get rid of this null check and convert `ApplyConfigurationUpdate` into `TryApplyConfigurationUpdate`.  Although, we would probably wouldn't want the cost of a method call each time.

Reply:
Do you mean to change `void ApplyConfigurationUpdate()` to something like `bool TryApplyConfigurationUpdate()`?

Ideally, I'd have to split the method to an inlinable `bool TryApplyConfigurationUpdate()` and a noinline `void ApplyConfigurationUpdate(ZeroLogConfiguration config)`, but I'm not sure if that's worth the additional complexity.

We'd pay the cost of a method call and the cost of an interlocked exchange, but that would be right before spinning. OTOH, we'd do this in a hot loop on every iteration, so I think I still prefer a standard null check, which I suppose will translate to a lower CPU usage.

Reply:
> Do you mean to change `void ApplyConfigurationUpdate()` to something like `bool TryApplyConfigurationUpdate()`?
>
> Ideally, I'd have to split the method to an inlinable `bool TryApplyConfigurationUpdate()` and a noinline `void ApplyConfigurationUpdate(ZeroLogConfiguration config)`, but I'm not sure if that's worth the additional complexity.

That's exactly what I meant.
 
> We'd pay the cost of a method call and the cost of an interlocked exchange, but that would be right before spinning. OTOH, we'd do this in a hot loop on every iteration, so I think I still prefer a standard null check, which I suppose will translate to a lower CPU usage.

We wouldn't pay the cost of a method call if the method was split and the part that calls `Interlocked.Exchange` was marked as inlineable.  However we would be left with the cost of the `xchg` on top of the `cmp`.  It's possibly not worth the complication however are we sure we don't need any memory fences when reading `_nextConfig`?

Reply:
Ok, I updated the code to only check for config updates on yield points, that should get rid of the cost issue.